waspc/src/Wasp/AppSpec/App/Client.hs

```diff
@@ -10,6 +10,7 @@ import Wasp.AppSpec.ExtImport (ExtImport)
 
 data Client = Client
   { setupFn :: Maybe ExtImport,
-    rootComponent :: Maybe ExtImport
+    rootComponent :: Maybe ExtImport,
+    baseDir :: Maybe String
```

Comment:
It would be nice if this was something like "Url" instead of "String", but probably best not to get into it now :D.
Although hm, what do we know about this String, about what it should look like? I guess it needs to be absolute path and that is it? We don't allow stuff like `bla` (relative) or `/bla?x=1` (query params) or `/bla#summary`, right? If that is so, maybe it might make sense to actually roll with StrongPath here that is Abs Dir! I can't remember if we can embed it directly into our AppSpec though, will our TemplateHaskell magic know how to translate this into Wasp DSL.

What I would do is maybe give it a try -> try to make it StrongPath Abs Dir -> if that works out of the box, great, we got some extra safety here. If not, forget it, but you can maybe put a comment next to it explaining what we expect it to look like.

If this works, you can also remove that validation you have later.

Reply:
> will our TemplateHaskell magic know how to translate this into Wasp DSL.
Sadly, it doesn't seem to be the case 🤷 
```
No translation to wasp type for type AppT (AppT (AppT (ConT StrongPath.Internal.Path) (ConT StrongPath.Internal.Posix)) (ConT StrongPath.Internal.Abs)) (ConT StrongPath.Internal.Dir')
```

> If this works, you can also remove that validation you have later.

If the validation fails, the error will not be something specific, but rather a generic StrongPath Abs Dir parsing error which will confuse users IMHO

Reply:
Ah ok, so we don't have a translation for that in our Evaluator.

I think if we had that working properly for StrongPath, that error toward user could be nice, it would basically tell them something like "this should be an absolute path to a dir and it is not", so I don't think it would confuse users.

But ok, since we don't have it right now, let's roll with validation check!